### PR TITLE
server, tablecodec: show region key's record id or index values.

### PR DESCRIPTION
**Pull request by disksing**
Add some details for `regionHandler`. If the key is a record key, show the record id. If the key is a index key, show the index values.

### CODE REVIEW

File: server/http_handler.go [hunk 1]
```diff
@@ -773,6 +775,13 @@ func NewFrameItemFromRegionKey(key []byte) (frame *FrameItem, err error) {
 	frame = &FrameItem{}
 	frame.TableID, frame.IndexID, frame.IsRecord, err = tablecodec.DecodeKeyHead(key)
 	if err == nil {
+		if frame.IsRecord {
+			_, frame.RecordID, err = tablecodec.DecodeRecordKey(key)
+		} else {
+			_, _, frame.IndexValues, err = tablecodec.DecodeIndexKey(key)
```
Comment (AndreMouche) [inline server/http_handler.go]: Could we get recordID from the no-unique index?

File: tablecodec/tablecodec.go [hunk 2]
```diff
@@ -109,6 +111,41 @@ func DecodeRecordKey(key kv.Key) (tableID int64, handle int64, err error) {
...
+	key = key[len(indexPrefixSep):]
+
+	key, indexID, err = codec.DecodeInt(key)
+	if err != nil {
+		return 0, 0, nil, errors.Trace(err)
+	}
+
+	for len(key) > 0 {
+		remain, d, e := codec.DecodeOne(key)
+		if err != nil {
```
Comment (AndreMouche) [inline tablecodec/tablecodec.go]: if e != nil

File: server/http_handler.go (hunk 1 above)
Comment (disksing) [reply server/http_handler.go]: It's hard to know the type of the index here. I think the best we can do it to show the recordID in index values (the last item).

File: tablecodec/tablecodec.go [hunk 3]
```diff
@@ -109,6 +111,41 @@ func DecodeRecordKey(key kv.Key) (tableID int64, handle int64, err error) {
...
+
+	if !key.HasPrefix(indexPrefixSep) {
+		return 0, 0, nil, errInvalidIndexKey.Gen("invalid index key - %q", k)
+	}
+	key = key[len(indexPrefixSep):]
+
+	key, indexID, err = codec.DecodeInt(key)
+	if err != nil {
+		return 0, 0, nil, errors.Trace(err)
+	}
```
Comment (zimulala) [inline tablecodec/tablecodec.go]: Could we replace line116-136 with `DecodeKeyHead`?

File: server/http_handler.go (hunk 1 above)
Comment (zimulala) [reply server/http_handler.go]: Did we write `IndexValues` into the response?

File: tablecodec/tablecodec_test.go [hunk 4]
```diff
@@ -323,3 +324,35 @@ func (s *testTableCodecSuite) TestReplaceRecordKeyTableID(c *C) {
...
 }
+
+func (s *testTableCodecSuite) TestDecodeIndexKey(c *C) {
+	tableID := int64(4)
+	indexID := int64(5)
+	values := []types.Datum{
+		types.NewIntDatum(1),
+		types.NewBytesDatum([]byte("abc")),
+		types.NewFloat64Datum(123.45),
+		// FIXME: MysqlTime is saved as uint64 internally. Without the schema
```
Comment (zimulala) [inline tablecodec/tablecodec_test.go]: Do we need to add some similar comments to `http_handler.go`?

**Comment by disksing** [comment]
PTAL @AndreMouche @zimulala

File: server/http_handler.go [hunk 5]
```diff
@@ -744,6 +746,13 @@ func NewFrameItemFromRegionKey(key []byte) (frame *FrameItem, err error) {
...
+		}
+		// Ignore decode errors.
+		err = nil
```
Comment (AndreMouche) [inline server/http_handler.go]: The err would always be nil in this function, so could we remove the err in response? Could we print some warning if any error happens?

Comment (disksing) [reply server/http_handler.go]: ok. I'm going to add an warning here.

**Pull request merged**